core/src/test/scala/org/apache/spark/deploy/history/HistoryServerSuite.scala

```diff
@@ -349,6 +349,17 @@ class HistoryServerSuite extends SparkFunSuite with BeforeAndAfter with Matchers
 
   }
 
+  /**
+   * Verify that the security manager needed for the history server can be instantiated
+   * even if spark.authenticate is set
+   */
+  test("SecurityManagerStartsWithSecureShuffle") {
+    val conf = new SparkConf()
+        .set("spark.testing", "true")
+        .set(SecurityManager.SPARK_AUTH_CONF, "true")
+    HistoryServer.createSecurityManager(conf)
```

Comment:
done. You know, this test could be merged into `"incomplete apps get refreshed"`, just by setting the security option and mentioning why it matters in a comment. I've patched that test to also use `HistoryServer.createSecurityManager` for consistency. it would save a fractional amount of test time